src/Streamly/Internal/Data/Parser/ParserD/Tee.hs

```diff
@@ -433,11 +474,35 @@ longest (Parser stepL initialL extractL) (Parser stepR initialR extractR) =
 
     where
 
+
     {-# INLINE_LATE initial #-}
     initial = do
-        sL <- initialL
-        sR <- initialR
-        return $ TeePair ([], StepState sL, [], []) ([], StepState sR, [], [])
+        resL <- initialL
+        resR <- initialR
+        return $ case resL of
+            IPartial sl ->
+                case resR of
+                     IPartial sr -> IPartial $ TeePair ([], StepState sl, [], [])
+                                                       ([], StepState sr, [], [])
+                     IDone br -> IPartial $ TeePair ([], StepState sl, [], [])
+                                                    ([], StepResult br, [], [])
+                     IError _ ->
+                         IPartial $ TeePair ([], StepState sl, [], [])
+                                            ([], StepResult undefined, [], [])
+            IDone bl ->
+                case resR of
+                     IPartial sr ->
+                         IPartial $ TeePair ([], StepResult bl, [], [])
+                                            ([], StepState sr, [], [])
+                     IDone _ -> IDone bl
+                     IError _ -> IDone bl
+            IError _ ->
+                case resR of
+                     IPartial sr ->
+                         IPartial $ TeePair ([], StepResult undefined, [], [])
```

Comment:
isn't it better to have separate states instead of using `undefined`

Reply:
Yeah, you're right. What I've written is incorrect.

Reply:
I'm going to leave it like so. It's marked as broken anyway. Don't see any point in investing time in it.